packages/reporters/hmr-server/src/HMRReporter.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+import {Reporter} from '@parcel/plugin';
+import HMRServer from './HMRServer';
+
+let hmrServer: HMRServer | null = null;
+export default new Reporter({
+  async report(event, options) {
+    if (!options.hot) return;
+
+    if (!hmrServer) {
+      hmrServer = new HMRServer();
+
+      if (options.hot === true) {
+        options.hot = {
+          port: 0,
+          certificateDir: '.parcel-cert'
+        };
+      }
+
+      await hmrServer.start(options.hot);
+    }
+
+    if (event.type === 'buildProgress' && event.phase === 'transformFinished') {
+      hmrServer.addChangedAsset(event.cacheEntry);
```

Comment:
I wonder if instead of doing this as a reporter event, we should just mark the changed assets in the graph in core and provide a list of them to the buildSuccess event instead? One fewer event, and less internal types to expose (CacheEntry).

Reply:
I think the many events actually causes the cli to crash for me so that would probably help also would be a lot faster